common/datablocks/src/kernels/data_block_sort.rs

```diff
@@ -87,7 +87,10 @@ impl DataBlock {
             .columns()
             .iter()
             .zip(rhs.columns().iter())
-            .map(|(a, b)| DataColumnCommon::merge_columns(a, b, indices))
+            .map(|(a, b)| {
+                let _ = &indices;
```

Comment:
This line used for?

Reply:
> This line used for?

see https://doc.rust-lang.org/nightly/edition-guide/rust-2021/disjoint-capture-in-closures.html

Auto gen by `cargo fix`. This is a conservative analysis: in many cases, these dummy lets can be safely removed and your program will work fine.

Reply:
It doesn't look like a good style : (

Reply:
Maybe we can do it with macros? similar to '[&captures_var] (args) {} in C + +`.

Reply:
> Maybe we can do it with macros? similar to '[&captures_var] (args) {} in C++`.

We can try to remove these dummy let manually. If it is really necessary, consider other options.